### DISCUSSION TITLE
Switching from Oracle JDK to the Microsoft Build of OpenJDK

### DISCUSSION THREAD

**Discussion by nedone15** [microsoft/openjdk#619] [Q&A] [answered]
By installing Oracle JDK I can quickly arrive at a situation on a Windows 10 PC where I can double click on a test jar file and the jar file opens.
I have installed the Microsoft Build of OpenJDK on a fresh Windows 11 PC (different from the above Win 10 PC) and after the install when I double click on the same test jar file, nothing happens.

--- Details of Oracle JDK setup
-- Among the Oracle JDK installation files available here
https://www.oracle.com/java/technologies/downloads/
I installed JDK 8 (that is, both the jdk-8u411-windows-i586.exe file and the jdk-8u411-windows-x64.exe file which come bundled together with the corresponding JRE) and on top of those two files I also installed JDK 11 (that is, the jdk-11.0.23_windows-x64_bin.exe file). After the installation of these three Oracle JDK installation files I just need to reboot the PC and all is ready for me: proper Windows path is set, association with jar and other Java files, etc.

--- Details of jar test file
-- As a publicly available test jar file I used a Java tool which is available here
https://ftp.pride.ebi.ac.uk/pub/databases/pride/resources/tools/submission-tool/latest/desktop/px-submission-tool.zip.
This zip file, once unzipped, contains a folder which has inside (among the others) the px-submission-tool-2.7.3.jar file which (on a Windows 10 PC with the three above Oracle JDKs installed) works straight out of the box and it opens by simply double clicking on it

--- Details of the Microsoft Build of OpenJDK setup
-- Trying to replicate the above Oracle JDK installation files, I used the Microsoft Build of OpenJDK available here
https://learn.microsoft.com/en-us/java/openjdk/download
and, precisely, the two JDK 8 x64 and x86 and the two JRE 8 x64 and x86 installation files available here
https://adoptium.net/temurin/releases/?version=8
plus the JDK 11 x64 installation file available here
https://aka.ms/download-jdk/microsoft-jdk-11.0.23-windows-x64.msi
-- Installing these five OpenJDK files on a fresh Windows 11 PC (different from the above Win 10 PC) and rebooting provided the proper Windows path and association with jar files but when I double click on the same test px-submission-tool-2.7.3.jar file, nothing happens

--- I think I might be simply missing a very simple step or setting to be able to double click on my test jar file and be able to open it with the Microsoft Build of OpenJDK on a fresh Windows 11 (or Windows 10 for what it matters).
Could someone help me finding out why my installation of the Microsoft Build of OpenJDK is not working straight out of the box like my installation of the Oracle JDK?

Thank you

Best Regards

**Comment by brunoborges** [comment]
Hi @nedone15,

It is a bit confusing why you have to install so many JDKs just to be able to run a single JAR file.

Is your goal to be able to run a JAR as an executable by double-clicking on it in Windows Explorer?

**Comment by nedone15** [comment]
Hi brunoborges,

Thanks for your feedback.

The are several reasons why I installed multiple JDK versions and I am OK with having those multiple JDK versions.

Yes: the aim in switching from Oracle JDKs to the Microsoft Builds of OpenJDK was being able to open a jar file by double clicking on it in Windows Explorer straight after the reboot after the installation of the JDKs.

This was always possible throughout years and years with Oracle JDKs but it didn’t work the first time I tried with the Microsoft Builds of OpenJDK as I detailed in my original message, while I was (maybe wrongly) expecting that that should be the default behavior also for the Microsoft Builds of OpenJDK.

Best Regards

**Reply by brunoborges** [reply]
> This was always possible throughout years and years with Oracle JDKs but it didn’t work the first time I tried with the Microsoft Builds of OpenJDK as I detailed in my original message, while I was (maybe wrongly) expecting that that should be the default behavior also for the Microsoft Builds of OpenJDK.

Indeed it hasn't been our intention to support this, but we are now looking into it.

**Comment by nedone15** [comment]
Is there a place where I could get to know when-and-if this feature will be implemented?
Thx

**Reply by brunoborges** [reply]
This very discussion is the place, at least for now.

**Comment by nedone15** [comment]
Got it.
What about eventual Microsoft-official workarounds for having a jar file being being opened by double click from the Microsoft Build of OpenJDK; are there any?
Thx

**Reply by brunoborges** [reply]
That's what we are looking into it.

**Comment by nedone15** [comment]
I thought there might have been a registry edit or something similar to perform on-the-fly while waiting that this feature will come automatically from the installation files. Anyways, either way will be useful.
I will wait to hear about this one: it would be useful if I could get news in this thread but I'll check generally in this forum.
Thx

**Comment by nedone15** [comment]
Hello,

I just noticed that the latest Oracle Java critical patch updates cycle happened yesterday.
I was wondering whether if, at the moment when those patches will eventually flow into the corresponding versions of the Microsoft Builds of OpenJDK, there will be any chance to also get some news about the eventual "double-clicking-on-a-jar-file feature/workaround" of this thread.

Thank you

Best Regards

**Reply by brianjstafford** [reply]
We are actively working on the updates to the Microsoft Build of OpenJDK and will be releasing those as soon as we can. We have no update on the other issue at this time.

**Comment by nedone15** [comment]
Hello,

Any chance that the "double-clicking-on-a-jar-file feature" of this thread might have been integrated into the latest release of the Microsoft Builds of OpenJDK which appeared yesterday?

Thank you

Best Regards

**Reply by gdams** [reply]
@nedone15 just to double check, are you enabling the `Associate .jar` feature when running the MSI installer?

**Reply by gdams** [reply]
when I download https://aka.ms/download-jdk/microsoft-jdk-11.0.23-windows-x64.msi and run the installer, I can open the jar file that you provided to reproduce this issue on Windows 11 by double-clicking on it

**Reply by gdams** [reply]
One other thing to consider is are you installing this as an admin user? or a non-admin user?

**Reply by nedone15** [reply]
Hi gdams, thanks for your reply.

\- With your "Associate .jar feature" do you mean one of the install options? If yes, in my initial text I wrote "... rebooting provided the proper Windows path and association with jar files..." so yes, *.jar files were associated to the Microsoft Build of OpenJDK. Can't recall which was the default association setting among the installation options during the installation, but yes, the association option was selected together with the installation option of adding the proper path to the Windows path.
\- I was admin when the installation happened and also after a cold boot when I tried double clicking on a jar file with no effect.

Now I'd like to repeat my test because I can't see why it's working for you and not for me: is there anything else on Windows 11 I need to be aware of for getting this right? Apart from being admin and being sure that the installation options of associating *.jar files to the Microsoft Build of OpenJDK and modifying the Windows path are both selected?

**Reply by nedone15** [reply]
Hi gdams,

I just tested again with https://aka.ms/download-jdk/microsoft-jdk-21.0.4-windows-x64.msi
\- Fresh Win 11 Pro 23H2 system (literally the first thing I do after having freshly installed the OS is launch the above installer)
\- I have admin rights

Installation options:
\- Install for "All users" is checked
\- "Modify PATH variable" is checked (as default)
\- "Associate .jar" is checked (as default)
\- "Set or override JAVA_HOME variable" is not checked (as default)
\- "JavaSoft (Oracle) registry keys" is not checked (as default)

After a warm boot (restart) I verify if the PATH variable has been modified and indeed I see there this "C:\Program Files\Microsoft\jdk-11.0.24.8-hotspot\bin" location.

Then:
\- When I double click on px-submission-tool-2.7.3.jar nothing happens
\- If I try to launch that jar file from CMD with for instance "java -jar px-submission-tool-2.7.3.jar" something happens

Can't really understand why for you double clicking on that same jar file is working and not for me.
Can anyone help?

Thank you

**Comment by nedone15** [comment, marked as answer]
I just realized that the java tool I was using as a test needs some specific ports to be opened to properly start and these ports were blocked by the firewall.
Once these ports were made reachable, everything worked fine and I could open the test java tool by double clicking on its jar file; so all is fine with the Microsoft Build of OpenJDK.
Thanks again for your feedback.
Bets Regards
